integrationTests/testInitializer.go

```diff
@@ -622,3 +647,497 @@ func extractUint64ValueFromTxHandler(txHandler data.TransactionHandler) uint64 {
+		}
+	}
+}
+
+type txArgs struct {
+	nonce    int
+	value    *big.Int
+	rcvAddr  []byte
+	sndAddr  []byte
+	data     string
+	gasPrice int
+	gasLimit int
+}
+
+func generateTx(
+	skSign crypto.PrivateKey,
+	signer crypto.SingleSigner,
+	args *txArgs,
+) *transaction.Transaction {
+	tx := &transaction.Transaction{
+		Nonce:    uint64(args.nonce),
+		Value:    args.value,
+		RcvAddr:  args.rcvAddr,
+		SndAddr:  args.sndAddr,
+		GasPrice: uint64(args.gasPrice),
+		GasLimit: uint64(args.gasLimit),
+		Data:     args.data,
+	}
+	txBuff, _ := TestMarshalizer.Marshal(tx)
+	tx.Signature, _ = signer.Sign(skSign, txBuff)
+
+	return tx
+}
+
+func skToPk(sk crypto.PrivateKey) []byte {
+	pkBuff, _ := sk.GeneratePublic().ToByteArray()
+	return pkBuff
+}
+
+// TestPrivateKeyHasBalance checks if the private key has the expected balance
+func TestPrivateKeyHasBalance(t *testing.T, n *TestProcessorNode, sk crypto.PrivateKey, expectedBalance *big.Int) {
+	pkBuff, _ := sk.GeneratePublic().ToByteArray()
+	addr, _ := TestAddressConverter.CreateAddressFromPublicKeyBytes(pkBuff)
+	account, _ := n.AccntState.GetExistingAccount(addr)
+	assert.Equal(t, expectedBalance, account.(*state.Account).Balance)
+}
+
+// GetMiniBlocksHashesFromShardIds returns miniblock hashes from body
+func GetMiniBlocksHashesFromShardIds(body dataBlock.Body, shardIds ...uint32) [][]byte {
+	hashes := make([][]byte, 0)
+
+	for _, miniblock := range body {
+		for _, shardId := range shardIds {
+			if miniblock.ReceiverShardID == shardId {
+				buff, _ := TestMarshalizer.Marshal(miniblock)
+				hashes = append(hashes, TestHasher.Compute(string(buff)))
+			}
+		}
+	}
+
+	return hashes
+}
+
+// EqualSlices checks if two slices are equal
+func EqualSlices(slice1 [][]byte, slice2 [][]byte) bool {
+	if len(slice1) != len(slice2) {
+		return false
+	}
+
+	//check slice1 has all elements in slice2
+	for _, buff1 := range slice1 {
+		found := false
+		for _, buff2 := range slice2 {
+			if bytes.Equal(buff1, buff2) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+
+	//check slice2 has all elements in slice1
+	for _, buff2 := range slice2 {
+		found := false
+		for _, buff1 := range slice1 {
+			if bytes.Equal(buff1, buff2) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
+
+// GeneratePrivateKeyInShardId generates a private key that is in a given shard
+func GeneratePrivateKeyInShardId(
+	coordinator sharding.Coordinator,
+	shardId uint32,
+) crypto.PrivateKey {
+
+	suite := kyber.NewBlakeSHA256Ed25519()
+	keyGen := signing.NewKeyGenerator(suite)
+	sk, pk := keyGen.GeneratePair()
+
+	for {
+		buff, _ := pk.ToByteArray()
+		addr, _ := TestAddressConverter.CreateAddressFromPublicKeyBytes(buff)
+
+		if coordinator.ComputeId(addr) == shardId {
+			return sk
+		}
+
+		sk, pk = keyGen.GeneratePair()
+	}
+}
+
+// GenerateRandomHexAddressInShard generates a random address in the given shard
+func GenerateRandomHexAddressInShard(
```

Comment:
We have this functionality in testWalletAccount.initCrypto. Maybe extract in a func?

Reply:
Done.